### ISSUE TITLE
Issue with CSV ingestion with non-default delimiter

### ISSUE THREAD

**Issue by mayankshriv** [apache/incubator-pinot#7187]
The code assumes that the delimiter for CSV in the reader config should also be used when specifying the header string in the ingestion spec. While this is (arguably) a valid assumption, it might be non intuitive for the user since these are two different configs in two different places.

Filing this issue for a better user experience, perhaps:
- If the CSV delimiter does not work for header String in job spec, try `comma`?
- At the very least, error out notifying the user that the two are inconsistent.

**Comment by suddendust** [comment]
I can pick this up @mayankshriv

**Comment by mayankshriv** [comment]
Thanks @suddendust, assigned to you. Let's first figure out the right approach, should we auto try a few separators, or gracefully error out with good information? First approach has the danger of picking the wrong delimiter. And the second one requires user to make a string with delimiters (that could be control characters) to specify in a text file.

Let's solicit some inputs here, and then go for an implementation.

**Comment by suddendust** [comment]
@mayankshriv thanks for assigning the story. So just to clarify, the record reader config takes in the delimiter for CSV as defined [here](https://docs.pinot.apache.org/basics/data-import/pinot-input-formats#:~:text=between%20the%20formats.-,CSV,-dataFormat%3A%20%27csv%27). In the [ingestion spec](https://docs.pinot.apache.org/configuration-reference/job-specification), what header are you referring to exactly?

**Comment by mayankshriv** [comment]
@suddendust: In the ingestion job spec yaml file, we provide `delimiter`, `header` as two configs. It is implicitly assumed that `header` strings will also be separated by `delimiter`.

**Comment by suddendust** [comment]
>First approach has the danger of picking the wrong delimiter. 

Yes, and the bad thing about this is that this can silently keep working (incorrectly) in the background. The max we can do in this approach is warn the user.

**Comment by suddendust** [comment]
@mayankshriv regarding

>And the second one requires user to make a string with delimiters (that could be control characters) to specify in a text file

Is this not the case even now? Say the header has a a control character as the delimiter, he will face the same problem even now right?

Also, how about accepting a different config for the header delimiter? This can be optional, in which case we can fallback to using the currently supplied delimiter, and throw an exception if it doesn't work. I think this is better than the application assuming the delimiters by itself and giving the user a surprise at some later point.

**Comment by suddendust** [comment]
@mayankshriv let me know what you think of this.

**Comment by mayankshriv** [comment]
I think the code and user point-of-view are contradicting here. It is very natural for users to provide a comma separated header string. May be we should gracefully error out right now if we don't find the delimiter in the header?

**Comment by suddendust** [comment]
That will be a good starting step for sure. However, can we also have an optional config that the user can set explicitly if the header delimiter is different than the CSV delimiter? If this is absent, we'll just keep using the existing config, and will error out if they're incompatible. It will be absent most of the time, because as you said, it's for users to provide a comma separated header string. But in cases where it isn't, this config will come in handy @mayankshriv.

**Comment by suddendust** [comment]
@mayankshriv I think gracefully erroring out if they don't match is the least we can do here. Shall we start with that?

**Comment by suddendust** [comment]
@mayankshriv Have raised a preliminary PR [here](https://github.com/apache/pinot/pull/7237)
